### PR TITLE
Page Structure and more UI Fundamentals

### PR DESCRIPTION
**Pull request by Chadtech**
In this PR:

- I made a pages module, that represents the different pages of the frontend (which, I imagine right now, should include the title page, a start game page, and a page for the actual game play).

- Basic UI components, including arranging things in a gird and a button.

### CODE REVIEW

File: ui/src/main.rs [hunk 1]
```diff
@@ -1,33 +1,47 @@
...
+    model: Model,
+}
+
+enum Model {
+    Title(title::Model),
 }
 
 enum Msg {
-    AddOne,
+    TitleMsg(title::Msg),
```
Comment (Chadtech) [inline ui/src/main.rs:24]: The main `Msg` type wraps the individual page `Msg` types

File: ui/src/page/title.rs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+// View //
+////////////////////////////////////////////////////////////////
+
+pub fn view(_model: Model) -> Vec<Element<Msg>> {
+    row::many(vec![
+        row(vec![cell(vec![text("FIGHT LINES")])]).center(),
+        row(vec![cell(vec![text("THE GAME")])]).center(),
+    ])
+    .view()
+}
```
Comment (Chadtech) [inline ui/src/page/title.rs:35]: The title page, with its own `Model`, `Msg`, `init` and `view` parts. It should eventually get an `update` too.

File: ui/src/page/title.rs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+////////////////////////////////////////////////////////////////
+// View //
+////////////////////////////////////////////////////////////////
+
+pub fn view(_model: Model) -> Vec<Element<Msg>> {
+    row::many(vec![
+        row(vec![cell(vec![text("FIGHT LINES")])]).center(),
+        row(vec![cell(vec![text("THE GAME")])]).center(),
+    ])
```
Comment (Chadtech) [inline ui/src/page/title.rs:33]: It looks kind of like this
```
                           FIGHT LINES                           |
                             THE GAME                            |
```
two rows, each with text on them, with their content centered. Thats what this code is building.

Comment (VictorXLR) [reply ui/src/page/title.rs:33]: Im ok with this

**Review by VictorXLR — APPROVED** [review]

**Review by VictorXLR — APPROVED** [review]
nice work

**Pull request merged**